src/idl_gen_cpp.cpp

```diff
@@ -2941,10 +2943,16 @@ class CppGenerator : public BaseGenerator {
           }
           case BASE_TYPE_UTYPE: {
             value = StripUnionType(value);
-            code += "_fbb.CreateVector<uint8_t>(" + value +
-                    ".size(), [](size_t i, _VectorArgs *__va) { "
-                    "return static_cast<uint8_t>(__va->_" +
-                    value + "[i].type); }, &_va)";
+            const auto &type = opts_.scoped_enums
```

Comment:
Let's simplify this line too as line 933 (UnionVectorVerifySignature) does. The lifetime extension isn't needed here.

Reply:
Updated!